network/linux_socket/linux_socket.c

```diff
@@ -249,14 +247,14 @@ static int32_t linux_socket_accept(struct linux_desc *desc, uint32_t sock_id,
 
 	ret = accept(sock_id, NULL, NULL);
 
-	flags = fcntl(sock_id, F_GETFL);
```

Comment:
Maybe [accept4](https://man7.org/linux/man-pages/man2/accept.2.html) could also be used here with the `SOCK_NONBLOCK` flag to make it non-blocking as soon as accepted.